packages/pharos/src/components/sheet/pharos-sheet.ts

```diff
@@ -0,0 +1,367 @@
+ * @slot footer - Contains the content of the sheet footer.
+ *
+ * @fires pharos-sheet-open - Fires when the sheet is about to open - cancelable
+ * @fires pharos-sheet-opened - Fires when the sheet has opened
+ * @fires pharos-sheet-close - Fires when the sheet is about to close - cancelable
+ * @fires pharos-sheet-closed - Fires when the sheet has closed
+ *
+ */
+export class PharosSheet extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-button': PharosButton,
+    'pharos-heading': PharosHeading,
+    'focus-trap': FocusTrap,
+  };
+
+  /**
+   * Indicates if the sheet is open.
+   * @attr open
+   */
+  @property({ type: Boolean, reflect: true })
+  public open = false;
+
+  /**
+   * Indicates if the sheet is expanded.
+   * @attr expanded
+   */
+  @property({ type: Boolean, reflect: true })
+  public expanded = false;
+
+  /**
+   * Indicates if the sheet is allowed to expand.
+   * @attr enableExpansion
+   */
+  @property({ type: Boolean, reflect: true })
+  public enableExpansion = true;
+
+  /**
+   * Indicates if the sheet contains close button.
+   * @attr hasClose
+   */
+  @property({ type: Boolean, reflect: true })
+  public hasClose = true;
+
+  /**
+   * Text content for the sheet header
+   * @attr header
+   */
+  @property({ type: String, reflect: true })
+  public header = 'Sheet header';
+
+  /**
+   * Indicates if the sheet footer should contain a divider.
+   * @attr footer-divider
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'footer-divider' })
+  public footerDivider = false;
+
+  private _currentTrigger: Element | null = null;
+
+  private _triggers!: NodeListOf<HTMLElement>;
+
+  @state()
+  private _isFooterEmpty = true;
+
+  private _startHeight = 0;
+  private _startY = 0;
+  private _isDragging = false;
+
+  constructor() {
+    super();
+    this._handleKeydown = this._handleKeydown.bind(this);
+    this._handleTriggerClick = this._handleTriggerClick.bind(this);
+    if (this.enableExpansion) {
+      this.addEventListener('touchend', this._handleDragEnd);
+      this.addEventListener('mouseup', this._handleDragEnd);
+      this.addEventListener('touchmove', this._handleTouchDragging);
+      this.addEventListener('mousemove', this._handleMouseDragging);
+    }
+  }
+
+  public static override get styles(): CSSResultArray {
+    return [sheetStyles];
+  }
+
+  protected override firstUpdated(): void {
+    this._addTriggerListeners();
+  }
+
+  protected override update(changedProperties: PropertyValues): void {
+    super.update && super.update(changedProperties);
+  }
+
+  protected override updated(changedProperties: PropertyValues): void {
+    if (changedProperties.has('open')) {
+      const body = document.querySelector('body');
+
+      if (this.open) {
+        body?.classList.add('pharos-sheet__body');
+        this._focusContents();
+      } else {
+        body?.classList.remove('pharos-sheet__body');
+        this._returnTriggerFocus();
```

Comment:
Is there a reason we use the `pharos-` prefix here?

Reply:
actually I don't think it is used. deleted.